diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/IndentationRule.kt

```diff
@@ -560,6 +574,60 @@ class IndentationRule(configRules: List<RulesConfig>) : DiktatRule(
             get() =
                 SPACE.toString().repeat(n = this)
 
+        /**
+         * @return `true` if the indent should be incremented once this node is
+         *   encountered, `false` otherwise.
+         * @see isIndentDecrementing
+         */
+        private fun ASTNode.isIndentIncrementing(): Boolean =
+            when (elementType) {
+                /*
+                 * This is a special case of an opening parenthesis which *may* or
+                 * *may not* be indent-incrementing.
+                 */
+                LPAR -> isParenthesisAffectingIndent()
+
+                else -> elementType in increasingTokens
+            }
+
+        /**
+         * @return `true` if the indent should be decremented once this node is
+         *   encountered, `false` otherwise.
+         * @see isIndentIncrementing
+         */
+        private fun ASTNode.isIndentDecrementing(): Boolean =
+            when (elementType) {
+                /*
+                 * This is a special case of a closing parenthesis which *may* or
+                 * *may not* be indent-decrementing.
+                 */
+                RPAR -> isParenthesisAffectingIndent()
+
+                else -> elementType in decreasingTokens
+            }
+
+        /**
+         * Parentheses only affect indent when they're a part of a
+         * [VALUE_PARAMETER_LIST] (formal arguments) or a [VALUE_ARGUMENT_LIST]
+         * (effective function call arguments).
+         *
+         * When they're children of a [PARENTHESIZED] (often inside a
+         * [BINARY_EXPRESSION]), they don't contribute to the indent.
+         *
+         * @return whether this [LPAR] or [RPAR] node affects indent.
+         * @see BINARY_EXPRESSION
+         * @see PARENTHESIZED
+         * @see VALUE_ARGUMENT_LIST
+         * @see VALUE_PARAMETER_LIST
+         */
+        private fun ASTNode.isParenthesisAffectingIndent(): Boolean {
+            require(elementType in arrayOf(LPAR, RPAR)) {
+                elementType.toString()
+            }
+
+            return treeParent.elementType != PARENTHESIZED
```

Comment:
How will we handle this case:
```kotlin
fun f() = (
    1 + 2
)
```
I think, `(1+2)` is a `PARENTHESIZED` node here, but we need to increase indent after `LPAR`

Reply:
You're right indeed, thanks.

I've added unit tests (a88d8ea6181daf16946a74a933e1a85a579ed5dd) to cover this case, will fix them soon.

Reply:
Fixed with a848a3e8eb479827de62c668818864c507facc10.